query/aql_processor.go

```diff
@@ -1321,6 +1321,24 @@ func (qc *AQLQueryContext) runBatchExecutor(e BatchExecutor, isLastBatch bool) {
 
 // copyHostToDevice copy vector party slice to device vector party slice
 func copyHostToDevice(vps memCom.HostVectorPartySlice, deviceVPSlice deviceVectorPartySlice, stream unsafe.Pointer, device int) (bytesCopied, numTransfers int) {
+	if memCom.IsArrayType(vps.ValueType) {
+		// for array data type
+		// copy offset-length
+		cgoutils.AsyncCopyHostToDevice(
+			deviceVPSlice.basePtr.getPointer(), vps.Offsets, vps.Length*8,
```

Comment:
we can add a device pointer (offsets) instead of directly use basePtr for readability although it is the same as basePtr.